### PR TITLE
Fix azure build by pinning cctbx_project

### PR DESCRIPTION
**Pull request by ndevenish**
Current SEGV is occuring due to a mismatch in the version used for compiling dxtbx (checked out into `modules/`) and the version used for python importing (from `lib/python3.X/site-packages/scitbx` in the conda environment).

This reads the cctbx-base version from the package list so that `bootstrap.py update` can be run before creating the conda environment, and fixes the branch to that.

Fixes https://github.com/cctbx/dxtbx/issues/383 and eventually fixesques https://github.com/xia2/xia2/issues/601, although latter may need xia2 change also.

### CODE REVIEW

File: .azure-pipelines/bootstrap.py [hunk 1]
```diff
@@ -671,12 +671,6 @@ def update_sources(options):
             "dials/cbflib",
         )
     }
-    repositories["cctbx_project"] = {
-        "base-repository": "cctbx/cctbx_project",
-        "effective-repository": "dials/cctbx",
-        "branch-remote": "master",
-        "branch-local": "stable",
-    }
```
Comment (Anthchirp) [inline .azure-pipelines/bootstrap.py:679]: this needs to be left in, otherwise it won't know what to check out

Comment (Anthchirp) [reply .azure-pipelines/bootstrap.py:679]: or maybe I have this wrong. Ignore pls.

Comment (ndevenish) [reply .azure-pipelines/bootstrap.py:679]: Yeah, it first creates the entry then overwrites it!

Comment (Anthchirp) [reply .azure-pipelines/bootstrap.py:679]: but it used to be harmless to leave in. Does it have to go because we now need a branch from the upstream remo?

**Pull request merged**